nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/AbstractCSVRecordReader.java

```diff
@@ -132,6 +131,10 @@ protected final Object convertSimpleIfPossible(final String value, final DataTyp
         return value;
     }
 
+    private String trim(String value) {
+        return value.startsWith("\"") && value.endsWith("\"") && (value.length() > 1) ? value.substring(1, value.length() - 1) : value;
```

Comment:
Doesn't seem like it would offer a significant performance improvement on average, only if most of the strings to be trimmed were single characters. I'm +1 on this one if you are, I can do the merge (to get this into 1.8.0 RC3) if you like.